examples/data-objects/task-selection/src/containerCode.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { BaseContainerRuntimeFactory } from "@fluidframework/aqueduct";
+import { IContainerRuntime } from "@fluidframework/container-runtime-definitions";
+import { RuntimeRequestHandler } from "@fluidframework/request-handler";
+import { RequestParser, requestFluidObject } from "@fluidframework/runtime-utils";
+
+import { OldestClientDiceRoller, OldestClientDiceRollerInstantiationFactory } from "./oldestClientDiceRoller";
+import { TaskManagerDiceRoller, TaskManagerDiceRollerInstantiationFactory } from "./taskManagerDiceRoller";
+import { IDiceRoller } from "./interface";
+
+const registryEntries = new Map([
+    OldestClientDiceRollerInstantiationFactory.registryEntry,
+    TaskManagerDiceRollerInstantiationFactory.registryEntry,
+]);
+
+export const taskManagerDiceId = "taskManagerDice";
+export const oldestClientDiceId = "oldestClientDice";
+
+// Just a little helper, since we're going to request multiple objects.
+async function requestObjectStoreFromId<T>(request: RequestParser, runtime: IContainerRuntime, id: string) {
+    const fluidObjectRequest = RequestParser.create({
+        url: ``,
+        headers: request.headers,
+    });
+    return requestFluidObject<T>(
+        await runtime.getRootDataStore(id),
+        fluidObjectRequest);
+}
+
+const requestHandler: RuntimeRequestHandler =
+    async (request: RequestParser, runtime: IContainerRuntime) => {
+        const diceId = request.pathParts[0];
+        if (diceId === taskManagerDiceId || diceId === oldestClientDiceId) {
+            const dice = await requestObjectStoreFromId<IDiceRoller>(
```

Comment:
I hesitate to add new calls to deprecated APIs, especially in new demos if I can avoid it.  I can look at simplification here though (Maybe even add a new `rootDataObjectRequestHandler()` in `@fluidframework/request-handler` to allow requesting root DO's by ID?  Which would function similar to `innerRequestHandler()` but limit access to roots only?)

Reply:
Yep, I'd create something here and put it next to defaultRouteRequestHandler
innerRequestHandler ideally should not be used as it is essentially allows uncontrolled access to the guts of container, but some scoped accessor / helpers should be Ok.
The other pattern we should explore (and expose for easy of use) is the one where requests are coming to default root object that consults section of root directory that basically holds key -> handle mapping.
This would work for static list (like here - two entries), but also for dynamic cases ("external URI manager" - something we would need to build to properly do external reference tracking in presence of GC).

Reply:
Done in latest